extensions/git/src/repository.ts

```diff
@@ -840,6 +841,17 @@ export class Repository implements Disposable {
 		return this._rebaseCommit;
 	}
 
+	private _mergeInProgress: boolean = false;
+
+	set mergeInProgress(value: boolean) {
+		this._mergeInProgress = value;
+		commands.executeCommand('setContext', 'gitMergeInProgress', value);
```

Comment:
Maybe we should avoid `setContext` calls whenever they aren't needed. We should probably do the same for the rebase one in another PR.

```suggestion
		if (this._mergeInProgress === value) {
				return;
		}

		this._mergeInProgress = value;
		commands.executeCommand('setContext', 'gitMergeInProgress', value);
```